File: frontend/src/types/comment.ts
export interface Comment {
  id: string
  body: string
  authorId: string
  authorName: string
  authorAvatarUrl: string | null
  personId: string | null
  storyId: string | null
  mediaId: string | null
  parentCommentId: string | null
  likesCount: number
  isLikedByMe: boolean
  createdAt: string
  updatedAt: string
  replies?: Comment[]
}

File: backend/app/http/schemas/comment.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class CommentCreate(BaseModel):
    body: str
    author_id: uuid.UUID
    person_id: uuid.UUID | None = None
    story_id: uuid.UUID | None = None
    media_id: uuid.UUID | None = None
    parent_comment_id: uuid.UUID | None = None


class CommentUpdate(BaseModel):
    body: str


class CommentResponse(BaseModel):
    id: uuid.UUID
    body: str
    author_id: uuid.UUID
    person_id: uuid.UUID | None
    story_id: uuid.UUID | None
    media_id: uuid.UUID | None
    parent_comment_id: uuid.UUID | None
    likes_count: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
